numpy/lib/function_base.py

```diff
@@ -4014,122 +4192,212 @@ def _quantile_is_valid(q):
     return True
 
 
-def _lerp(a, b, t, out=None):
-    """ Linearly interpolate from a to b by a factor of t """
-    diff_b_a = subtract(b, a)
-    # asanyarray is a stop-gap until gh-13105
-    lerp_interpolation = asanyarray(add(a, diff_b_a*t, out=out))
-    subtract(b, diff_b_a * (1 - t), out=lerp_interpolation, where=t>=0.5)
-    if lerp_interpolation.ndim == 0 and out is None:
-        lerp_interpolation = lerp_interpolation[()]  # unpack 0d arrays
-    return lerp_interpolation
+def _virtual_index_formula(array_size: np.array,
+                           quantiles: np.array,
+                           alpha: float,
+                           beta: float) -> np.array:
+    """
+   Compute the floating point indexes of an array for the linear
+   interpolation of quantiles.
+   Reference:
+   Hyndman&Fan paper "Sample Quantiles in Statistical Packages",
+   DOI: 10.1080/00031305.1996.10473566
+   """
+    return array_size * quantiles + (
+            alpha + quantiles * (1 - alpha - beta)
+    ) - 1
```

Comment:
Ok for `array_size` -> `n`

The `-1` should be outside the bracket. It is used to correct the index which otherwise would be for an array starting at 1 (like in R)

I personally like exposing the whole formula because it looks like the formula in the original paper, which IMO help understand what's going on.
I had to rewrite multiple times `_lerp` (now `_linear_interpolation_formula`) so I think I understand what you mean but I don't know if factorizing `quantile` will mess with the boundary.